compiler/noirc_frontend/src/tests/metaprogramming.rs

```diff
@@ -141,3 +141,23 @@ fn errors_if_macros_inject_functions_with_name_collisions() {
         ) if contents == "foo"
     ));
 }
+
+#[test]
+fn uses_correct_type_for_attribute_arguments() {
+    let src = r#"
+    #[foo(32)]
```

Comment:
I mean, is it expected to be set on itself, or is that just for the sake of testing?

Reply:
Oh, it's just for the sake of testing. I copied it from the snippet in https://github.com/noir-lang/noir/issues/6373 . Before that I would always use a different function to call at compile time, but this way there's less code 🤷

Reply:
> Does it evaluate foo at comptime with 32?

Yep. When you have an attribute like `#[a(b, c)] fn d(...) {}` all this does is call the function `a` with `d` as its first argument, then `b` and `c` as the other arguments. So the full call would look like `a(d, b, c)` if you called it outside of an attribute. Passing `d` is different than in "normal" code though since it passes d as a `FunctionDefinition` rather than a callable function value.

Reply:
Thanks! What confused me was that the attribute was on the function itself, like `#[a(b, c)] fn a(_f, b, c) { ... }`; it looked like the purpose could be something like you see in JavaScript sometimes with `(function(b, c))(1, 2)`. I thought we wanted to execute `foo` at that point to e.g. generate and splice some code, then discard `foo`. Now I understand it's a convenience to test without having to define another to apply `foo` to.